src/sinks/aws_cloudwatch_logs.rs

```diff
@@ -96,14 +121,74 @@ impl crate::topology::config::SinkConfig for CloudwatchLogsSinkConfig {
     }
 }
 
-impl CloudwatchLogsSvc {
+impl CloudwatchLogsPartitionSvc {
     pub fn new(config: CloudwatchLogsSinkConfig) -> Result<Self, String> {
+        Ok(Self {
+            config,
+            clients: HashMap::new(),
+        })
+    }
+}
+
+impl Service<PartitionInnerBuffer<Vec<Event>, CloudwatchKey>> for CloudwatchLogsPartitionSvc {
+    type Response = ();
+    type Error = Box<std::error::Error + Send + Sync + 'static>;
+    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error> + Send + 'static>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        Ok(().into())
+    }
+
+    fn call(&mut self, req: PartitionInnerBuffer<Vec<Event>, CloudwatchKey>) -> Self::Future {
+        let (events, key) = req.into_parts();
+
+        let timeout = self.config.request_timeout_secs.unwrap_or(60);
+        let in_flight_limit = self.config.request_in_flight_limit.unwrap_or(5);
+        let rate_limit_duration = self.config.request_rate_limit_duration_secs.unwrap_or(1);
+        let rate_limit_num = self.config.request_rate_limit_num.unwrap_or(5);
+
+        let svc = if let Some(svc) = &mut self.clients.get_mut(&key) {
+            svc.clone()
+        } else {
+            let svc = {
+                let cloudwatch = CloudwatchLogsSvc::new(&self.config).unwrap();
+                let timeout = Timeout::new(cloudwatch, Duration::from_secs(timeout));
+
+                // TODO: add Buffer/Retry here
+
+                let rate = RateLimit::new(
+                    timeout,
+                    Rate::new(rate_limit_num, Duration::from_secs(rate_limit_duration)),
+                );
+                let concurrency = ConcurrencyLimit::new(rate, in_flight_limit);
+
+                Buffer::new(concurrency, 5)
+            };
+
+            self.clients.insert(key, svc.clone());
```

Comment:
Do we ever clean these up? Worth thinking about how this will behave when left running forever.

Reply:
Yeah, this is something I brought up with ben, I'm not totally sure how to approach it...do we run a kinda like gc pass? I also think we can reduce the amount of spawned services to per group instead of per group/stream pair.

Reply:
I think we can do some napkin math to determine if it's actually a problem and then decide how much to invest in fixing it. I know a lot of the rusoto stuff is `Arc`'d down pretty far, so it's possible these services are small enough that it doesn't matter. But we should verify that one way or the other.

Reply:
Yeah, I think here we are fine since we really only need to keep the few strings around. The confing in each of these services is pretty small.